components/core/dock.tsx

```diff
@@ -0,0 +1,220 @@
+'use client';
+
+import {
+  motion,
+  MotionValue,
+  useAnimation,
+  useMotionValue,
+  useSpring,
+  useTransform,
+  type SpringOptions,
+} from 'framer-motion';
+import {
+  Children,
+  cloneElement,
+  createContext,
+  useContext,
+  useEffect,
+  useRef,
+} from 'react';
+import { cn } from '@/lib/utils';
+
+/** Constants */
+const DOCK_HEIGHT = 128;
+const DEFAULT_MAGNIFICATION = 80;
+const DEFAULT_DISTANCE = 150;
+/** End of constants */
+
+/** Types */
+type TDock = {
```

Comment:
use `ComponentProps`, (`DockProps` here for example) please

Reply:
I Will do it